crates/iota-sdk-types/src/type_tag/mod.rs

```diff
@@ -267,6 +300,76 @@ impl PartialEq<str> for Identifier {
     }
 }
 
+impl std::ops::Deref for Identifier {
+    type Target = IdentifierRef;
+    fn deref(&self) -> &IdentifierRef {
+        unsafe { std::mem::transmute::<&str, &IdentifierRef>(self.0.as_ref()) }
+    }
+}
+
+impl std::borrow::Borrow<IdentifierRef> for Identifier {
+    fn borrow(&self) -> &IdentifierRef {
+        self
+    }
+}
+#[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[repr(transparent)]
+pub struct IdentifierRef(str);
+
+impl IdentifierRef {
+    pub const fn const_new(s: &'static str) -> &'static Self {
+        if !Identifier::is_valid(s) {
+            panic!("String is not a valid Move identifier");
+        }
+        // SAFETY: the following transmute is safe because
+        // (1) it's equivalent to the unsafe-reborrow inside IdentStr::ref_cast()
+        //     (which we can't use b/c it's not const).
+        // (2) we've just asserted that IdentStr impls RefCast<From = str>, which
+        //     already guarantees the transmute is safe (RefCast checks that
+        //     IdentStr(str) is #[repr(transparent)]).
+        // (3) both in and out lifetimes are 'static, so we're not widening the
+        // lifetime. (4) we've just asserted that the IdentStr passes the
+        // is_valid check.
+        unsafe { std::mem::transmute::<&'static str, &'static Self>(s) }
+    }
+    /// Returns true if this string is a valid identifier.
+    pub fn is_valid(s: &str) -> bool {
+        Identifier::is_valid(s)
+    }
```

Comment:
Why have this? There isn't even a way to construct a IdentifierRef that would be invalid?